apis/helpers/namer_test.go

```diff
@@ -7,57 +7,74 @@ import (
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
+const (
+	shortSuffix = "deploy"
+	// hashLen is 10 to represent "-" + hash + "-" (hash itself is always 8 characters)
+	hashLen = 10
+)
+
 func TestGetName(t *testing.T) {
-	for i := 0; i < 10; i++ {
-		shortName := randSeq(rand.Intn(validation.DNS1123SubdomainMaxLength-1) + 1)
-		longName := randSeq(validation.DNS1123SubdomainMaxLength + rand.Intn(100))
+	// shortBase + "-" + hash will be underneath the maximum limit without any truncation
+	shortBase := randSeq(validation.DNS1123SubdomainMaxLength - hashLen + 1)
```

Comment:
Doesn't this still work if we continue to use `rand.Intn` but fix the math?
```suggestion
	shortBase := randSeq(rand.Intn(validation.DNS1123SubdomainMaxLength - hashLen) + 1)
```

(I may still be off by 1, but that's the idea.)

I gather the point is to make sure `shortBase` is never long enough to trigger hashing/truncation when used with a `shortSuffix`; and `longBase`/`longSuffix` are long enough to guarantee that that behavior _is_ triggered.

And the original randomness was there to improve coverage over many runs by using samples of varying lengths, as long as they were within the expected bounds.

So I think we still want that randomness; we just need to make sure we're guaranteeing the correct bounds.

Reply:
Oh it definitely would - I can definitely put `rand.Intn` back to keep that behavior. 

With this approach I think I should put back the loop to generate 10 random `shortBase`s per test run as well?

Reply:
Updated the commit message accordingly as well to remove the notes about removing `rand.Intn` and making it deterministic!